src/components/LayoutTree.tsx

```diff
@@ -343,7 +346,19 @@ const linearizeTree = (
     return [...accum, node, ...descendants]
   }, [])
 
-  return thoughts
+  // Generate isLastVisible property here because determinig the next thought is difficult in the recursive call. So we do it after tree of thought is completely generated.
+  const treeThoughtsWithIsLastVisible = thoughts.map((node, index, thought) => {
+    const nextNode = thought[index + 1]
+    const nextAutofocus = nextNode ? calculateAutofocus(state, nextNode.path) : null
```

Comment:
This needs to be written with a bit more performance sensitivity, as `calculateAutofocus` is a heavy function.

If you're going to call `calculateAutofocus` on every node in `linearizeTree`, you must as well store the value and pass it down the component hierarchy to avoid repeating the same calculation further down.

See our previous discussion here: https://github.com/cybersemics/em/pull/2381#discussion_r1759746404.

Reply:
So I guess you are saying it should be calculated once, instead of mapping through the `treeThought` array, and then assigned to the `treeThought` array elements. Then we should pass it down the hierarchy?

Reply:
I had to map the `treeThought` because it wasn't going to work out in `reduce` method, and since we have to calculate it for every thought I needed to map over the `treeThought` array to calculate it.

Reply:
Yes, that should work.